utils/misc.ml

```diff
@@ -203,6 +203,21 @@ module Stdlib = struct
         else if p (Array.unsafe_get a1 i) (Array.unsafe_get a2 i) then true
         else loop (succ i) in
       loop 0
+
+    let for_alli p a =
+      let n = Array.length a in
+      let rec loop i =
+        if i = n then true
+        else if p i (Array.unsafe_get a i) then loop (succ i)
+        else false in
+      loop 0
+
+    let all_somes a =
```

Comment:
I would prefer `map_opt` for this function, as an instance of the Haskell function

```haskell
mapM : Monad m => (a -> m b) -> t a -> m (t b)
```

which is also called `traverse` (in the Applicative case), less evocative here.

But this is `misc` so we can do as we feel, and it looks like you don't feel the same way.